spec/controllers/journal_entries_controller_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+RSpec.describe JournalEntriesController, type: :controller do
+  let(:journal) { FactoryBot.create(:journal) }
+  let(:entry) { FactoryBot.create(:journal_entry, journal: journal) }
+  let(:valid_params) { FactoryBot.attributes_for(:journal_entry) }
+
+  describe 'GET #index' do
+    let(:entry2) { FactoryBot.create(:journal_entry, journal: journal) }
+    it 'populates an array of all journal entries' do
+      get :index, params: { journal_id: journal.id, journal_entry: valid_params }
+      expect(assigns(:journal_entries)).to match_array [entry, entry2]
+    end
+
+    it 'redirects to the journals page' do
+      get :index, params: { journal_id: journal.id, journal_entry: valid_params }
+      expect(response).to redirect_to journal_path(journal) 
+    end
```

Comment:
```suggestion
    get :index, params: { journal_id: journal.id, journal_entry: valid_params }

    it 'populates an array of all journal entries' do
      expect(assigns(:journal_entries)).to match_array [entry, entry2]
    end

    it 'redirects to the journals page' do
      expect(response).to redirect_to journal_path(journal) 
    end
```

Reply:
please DRY this in other `describe` blocks as well

Reply:
I'd like to leave it like this just to have a reference for both of the options on how to test, if this is ok